style: apply /style-guide pass to models/integrations (#2673)
wandb/docs: models/integrations/cohere-fine-tuning.mdx

---

> description: "Fine-tune Cohere models with W&B experiment tracking to log training metrics and monitor model performance."

Reviewer: I checked the Cohere docs and the support has been deprecated. Do we still want to keep this page?

Author: Keeping the page for now per maintainer decision — removing it is out of scope for this style pass, so I'm leaving this thread open for a separate content call.

title: Cohere fine-tuning
keywords: ["command model", "rerank fine-tune", "Cohere train API"]
---

> With W&B, you can log your Cohere model's fine-tuning metrics and configuration to analyze your models' performance and share results with your colleagues. This page shows you how to connect a Cohere fine-tuning run to a W&B project so that W&B captures training and validation metrics, hyperparameters, and run metadata automatically in your workspace. It's intended for users who are already fine-tuning Cohere models and want centralized experiment tracking.

Reviewer: * Not sure where we landed on this, but I think we aim to not use "This page shows you how to..."
* Some passive voice that can be improved.
*
```suggestion
With W&B, you can log your Cohere model's fine-tuning metrics and configuration to analyze your model performance and share results with your colleagues. This page shows you how to connect a Cohere fine-tuning run to a W&B project so that W&B captures training and validation metrics, hyperparameters, and run metadata automatically in your workspace. Use this guide if you are already fine-tuning Cohere models and want centralized experiment tracking.
```

Author: Applied — dropped 'This page shows you how' and the passive voice.

For a full example of how to start a fine-tuning run, see the [Cohere fine-tuning guide](https://docs.cohere.com/page/convfinqa-finetuning-wandb), and refer to the [Cohere API reference for the `wandb` setting](https://docs.cohere.com/reference/createfinetunedmodel#request.body.settings.wandb).

## Log Cohere fine-tuning results

To add Cohere fine-tuning logging to your W&B workspace, do the following:

1. Create a `WandbConfig` with your W&B API key, W&B `entity`, and `project` name. The API key authenticates the Cohere job with W&B, and the entity and project determine where W&B logs your runs. Create an API key in your [W&B user settings](https://wandb.ai/settings). Replace `<wandb-api-key>` in the following example with your API key.

2. Pass this config to the `FinetunedModel` object along with your model name, dataset, and hyperparameters to start your fine-tuning run. The `wandb` setting configures Cohere to stream metrics to your W&B project during the run.

    ```python
    from cohere.finetuning import WandbConfig, FinetunedModel

    # create a config with your W&B details
    wandb_ft_config = WandbConfig(
        api_key="<wandb-api-key>",

>         entity="my-entity", # must be a valid enitity associated with the provided API key

Author: Fixed — `enitity`→`entity`.
